### DISCUSSION TITLE
Progress bar on macOS -- why no advcpmv compatibility?

### DISCUSSION THREAD

**Discussion by oliversturm** [jarun/nnn#1940] [General]
I built and installed `cpg` and `mvg` on macOS, right from https://github.com/jarun/advcpmv -- no problem. Tried running `nnn -r` and was very surprised that this did not show progress bars during file operations. Now I see [here](https://github.com/jarun/nnn/blob/e7b097455101377f5ede05c7966886acc8303786/src/nnn.c#L8699-L8702) that the commands are only used on Linux, and [this wiki section](https://github.com/jarun/nnn/wiki/Advanced-use-cases#cp-mv-progress) says clearly that the feature is Linux only.

What's missing is any sort of explanation for this random limitation. Why not use the commands if they are available? I fail to see a downside to that. 

What's more, I can't figure out what this pointer to the `^T` shortcut is supposed to mean (as discussed in #333) -- I can press this during the ongoing operation but all I see is a line like this:

```
load: 2.16  cmd: mv 49114 uninterruptible 0.00u 0.27s
```

Bit fiddly to do this all the time, and not exactly a progress bar (edit: to be clear, there's no progress information at all, and no significant update to the output if I do this multiple times). However, in case there are pointers to tell me how I'm pressing `^T` wrong, please let's see them. Meanwhile I might go ahead and build my own `nnn` without the Linux ifdef in place -- curious to hear whether there are reasons not to use advcpmv on macOS?

**Comment by oliversturm** [comment]
Quick update: with the revolutionary diff below this works fine on my Mac. Perhaps it would be better to replace the `__linux__` bit with something else in case there are caveats here that I'm not familiar with. Should I submit this as a PR? Would be nice to get it out of Homebrew just like that in the future without having to patch my own.

```diff
diff --git a/src/nnn.c b/src/nnn.c
index 1208288b..2c4e249e 100644
--- a/src/nnn.c
+++ b/src/nnn.c
@@ -8693,10 +8693,8 @@ int main(int argc, char *argv[])
 			g_state.forcequit = 1;
 			break;
 		case 'r':
-#ifdef __linux__
 			memcpy(cp, PROGRESS_CP, sizeof PROGRESS_CP);
 			memcpy(mv, PROGRESS_MV, sizeof PROGRESS_MV);
-#endif
 			break;
 		case 'R':
 			cfg.rollover = 0;
```

**Comment by jarun** [comment]
One of the reasons would be I don't own a Mac myself and didn't consider enabling it.

If you can test and confirm things work smoothly we can enable it.

**Reply by oliversturm** [reply]
Thanks, will do. Let me run it a little while as I have it now and I'll let you know if I find any issue.

**Comment by oliversturm** [comment]
I have used the setup for a few weeks now and everything worked fine. However, today I was moving 180 files of varying medium to large sizes, and suddenly after  around 120 or so the process stopped and threw out errors like these for all the remaining files:

```
mvg: pipe: du: Too many open files
mvg: cannot create regular file './1.mkv': Too many open files
mvg: pipe: find: Too many open files
mvg: pipe: du: Too many open files
mvg: cannot create regular file './2.mkv': Too many open files
mvg: pipe: find: Too many open files
mvg: pipe: du: Too many open files
```

I have no idea where this comes from and it may be completely unrelated. I have now started the process again for the remaining files and it ran through without trouble. Just thought I'd mention it in case there's something to be learned here.

**Comment by jarun** [comment]
If your results are good, please raise a PR.
